internal/flex/flex.go

```diff
@@ -126,6 +126,15 @@ func FlattenFrameworkListString(ctx context.Context, l []string, diags *diag.Dia
 	return tfList
 }
 
+func FlattenFrameworkListStringWithNilAsEmpty(ctx context.Context, l []string, diags *diag.Diagnostics) types.List {
```

Comment:
what is different here from the function above? what is ` types.ListValueMust(types.StringType, nil)` necessary here?

Reply:
Default Realms when unset passed gives this "as cty.ListValEmpty(cty.String), but now null" particular error . Adding a Default value didn't solve the problem , so in my opinion changing the Flatten was the solution as done with strings previously . ListValueMust was used cause we needed an empty defaultRealms value , So listValueNull coudn't be used instead.

Reply:
so what you want here is that `[]string` should be flattened as `ListValEmpty` instead of `ListNull`. 

you don't need the `ListValueMust` here, you can simply remove the `if` block, and let `ListValueFrom` handle the empty list. See `FlattenFrameworkListInt64NotNull` below. The naming is quite inconsistent here though :)

Reply:
Modifier the function and removed the `if` block .